tools/integration_tests/util/creds_tests/creds.go

```diff
@@ -18,53 +18,87 @@ package creds_tests
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path"
+	"slices"
+	"strings"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/compute/metadata"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
 	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
 )
 
-const NameOfServiceAccount = "creds-test-gcsfuse"
+const NameOfServiceAccount = "creds-integration-tests"
+const CredentialsSecretName = "gcsfuse-integration-tests"
 
-func setPermission(permission string, serviceAccount string) {
-	// Provide permission to the bucket.
-	setup.RunScriptForTestData("../util/creds_tests/testdata/provide_permission.sh", setup.TestBucket(), serviceAccount, permission)
-}
+var WhitelistedGcpProjects = []string{"gcs-fuse-test", "gcs-fuse-test-ml"}
```

Comment:
What's the rationale behind adding this whitelisting logic?

Reply:
GCSFuse has only two GCP projects where test can be run. gcs-fuse-test is currently used for running CD tests and gcs-fuse-test-ml is used for CI tests.
If the tests are run on other projects, not the tests will not fail but a message will be printed in stdout that credentials tests are not run.